### PR TITLE
LoginController integration test to unit test

### PR DESCRIPTION
**Pull request by hedgehog1833**
For the login controller only two tests can easily be converted to unit tests. The tests that check for redirections etc. need more context and have to be integration tests - or at least I could not find out how to add the necessary handlers.
I also found out that we don't need the WithExceptionResolver interface as we can directly add the RestExceptionHandler class. I think this is because of the @ControllerAdvice annotation.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[IntegrationTests to UnitTests](https://trello.com/c/x6hnzTvO/121-integrationtests-to-unittests)

**Comment by hedgehog1833** [comment]
I removed the commented tests in the ArtistsRestControllerTests and did some minor refactorings.

File: webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java [hunk 1]
```diff
@@ -62,64 +57,99 @@ void tearDown() {
...
+        .andExpect(redirectedUrl(Endpoints.Guest.LOGIN + "?badCredentials"));
+
+    String expectedUsernameArgument = (username == null || username.isBlank()) ? "" : username;
+    verify(userService, times(1)).loadUserByUsername(expectedUsernameArgument);
+  }
+
+  @ParameterizedTest(name = "[{index}]: Username <{0}> and Password <{1}>")
+  @MethodSource("credentialProvider")
+  @DisplayName("Login with bad credentials should return call user service")
+  void login_with_bad_credentials_should_return_call_user_service(String username, String plainPassword) throws Exception {
```
Comment (DanielW1987) [inline webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java:129]:
```suggestion
  @DisplayName("Login with bad credentials return call user service")
  void login_with_bad_credentials_should_call_user_service(String username, String plainPassword) throws Exception {
```

File: webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java [hunk 2]
```diff
@@ -62,64 +57,99 @@ void tearDown() {
...
-  @DisplayName("Login with bad credentials should fail")
-  void login_with_bad_credentials_should_fail(String username, String plainPassword) throws Exception {
-    mockMvc.perform(post(Endpoints.Guest.LOGIN)
-                .with(SecurityMockMvcRequestPostProcessors.csrf())
-                .param(PARAM_USERNAME, username)
-                .param(PARAM_PASSWORD, plainPassword))
-            .andExpect(status().is3xxRedirection())
-            .andExpect(redirectedUrl(Endpoints.Guest.LOGIN + "?badCredentials"));
+  @DisplayName("Login with bad credentials should return redirect to login page")
+  void login_with_bad_credentials_should_return_redirect_to_login_page(String username, String plainPassword) throws Exception {
```
Comment (DanielW1987) [inline webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java:108]:
```suggestion
  @DisplayName("Login with bad credentials should redirect to login page")
  void login_with_bad_credentials_should_redirect_to_login_page(String username, String plainPassword) throws Exception {
```

File: webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java [hunk 3]
```diff
@@ -62,64 +57,99 @@ void tearDown() {
...
+
+    // when
+    var resultActions = mockMvc.perform(request);
+
+    // then
+    resultActions.andExpect(status().is3xxRedirection())
+        .andExpect(redirectedUrl(Endpoints.Guest.LOGIN + "?badCredentials"));
+
+    String expectedUsernameArgument = (username == null || username.isBlank()) ? "" : username;
+    verify(userService, times(1)).loadUserByUsername(expectedUsernameArgument);
```
Comment (DanielW1987) [inline webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java:123]: think you can remove that...

Comment (hedgehog1833) [reply webapp/src/test/java/rocks/metaldetector/web/controller/mvc/authentication/LoginControllerIT.java]: Sloppy, sorry 😉

**Pull request merged**